### PR TITLE
[BugFix] Fix the bug that the result of BoolLiteral getStringValue is inconsistent with the format of JAVA Boolean.parseBoolean

### LINKED ISSUE #12311: [User Defined Variable]error using boolean data type variable
### Steps to reproduce the behavior (Required)
1. 
```
set @'test_date_type'=(select cast(1 as boolean));
```
2. 
```
select @'test_date_type';
```

### Expected behavior (Required)
1
### Real behavior (Required)
0
image

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
```
+----------------------+
| current_version()    |
+----------------------+
| MAIN-RELEASE 37997b9 |
+----------------------+
```

### PR DESCRIPTION
**Pull request by HangyuanLiu**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #12311 #12360 #12356

## Problem Summary(Required) ：

The getStringValue of BoolLiteral does not meet expectations, and the bool of String format should return True and False. UserVariable will call JAVA Boolean.parseBoolean according to the result of getStringValue, if the incoming parameter is 1, it will return False, resulting in an incorrect result
## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 1 / 1 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/analysis/BoolLiteral.java|1|1|100.00%|[]|

**Review by packy92 — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/analysis/BoolLiteral.java [hunk 1]
```diff
@@ -112,7 +112,7 @@ public String toSqlImpl() {
 
     @Override
     public String getStringValue() {
-        return value ? "1" : "0";
+        return value ? "TRUE" : "FALSE";
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/analysis/BoolLiteral.java:115]: does this change will affect other places?   many places use this method.

**Comment by HangyuanLiu** [comment]
run starrocks_fe_unittest

**Comment by HangyuanLiu** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 1 / 1 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/analysis/BoolLiteral.java|1|1|100.00%|[]|

**Review by Seaven — APPROVED** [review]

**Review by kangkaisen — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Pull request merged**